spec/jobs/facilities/mental_health_reload_job_spec.rb

```diff
@@ -0,0 +1,116 @@
+    expect(facility2.local_updated).to be >= now
+  end
+
+  it 'deletes removed keys' do
+    Facilities::MentalHealthReloadJob.new.perform
+    expect(FacilityMentalHealth.find('101A')).to_not be_nil
+    expect(FacilityMentalHealth.find('202A')).to_not be_nil
+
+    allow_any_instance_of(
+      Facilities::MentalHealthReloadJob
+    ).to receive(:fetch_mental_health_data).and_return(CSV::Table.new([facility2]))
+
+    Facilities::MentalHealthReloadJob.new.perform
+    expect(FacilityMentalHealth.find('101A')).to be_nil
+    expect(FacilityMentalHealth.find('202A')).to_not be_nil
+  end
+
+  it 'updates modified data' do
+    now = Time.now.utc.iso8601
+    Facilities::MentalHealthReloadJob.new.perform
+    expect(FacilityMentalHealth.find('101A')).to_not be_nil
+    expect(FacilityMentalHealth.find('202A')).to_not be_nil
+    expect(FacilityMentalHealth.find('101A').mh_phone).to eq('407-123-1234')
+    expect(FacilityMentalHealth.find('202A').mh_phone).to eq('321-123-1234')
+    expect(FacilityMentalHealth.find('101A').local_updated).to be >= now
+    expect(FacilityMentalHealth.find('202A').local_updated).to be >= now
+    later = Time.now.utc.iso8601
+
+    # This data is the same as above EXCEPT for the phone number for 202A
+    mental_health_data = [
+      facility1,
+      CSV::Row.new(mental_phone_headers,
+                   ['202A', '321-987-6543', '0002', '2019-09-06T13:00:00.000'])
+    ]
+
+    allow_any_instance_of(
+      Facilities::MentalHealthReloadJob
+    ).to receive(:fetch_mental_health_data).and_return(CSV::Table.new(mental_health_data))
+
+    Facilities::MentalHealthReloadJob.new.perform
+    expect(FacilityMentalHealth.find('101A')).to_not be_nil
+    expect(FacilityMentalHealth.find('202A')).to_not be_nil
+    expect(FacilityMentalHealth.find('101A').mh_phone).to eq('407-123-1234')
+    expect(FacilityMentalHealth.find('202A').mh_phone).to eq('321-987-6543')
+    expect(FacilityMentalHealth.find('101A').local_updated).to be >= now
+    expect(FacilityMentalHealth.find('101A').local_updated).to be <= later
+    expect(FacilityMentalHealth.find('202A').local_updated).to be >= later
+  end
+
+  it 'cleans up bad extension data' do
+    mental_health_data = [
+      CSV::Row.new(mental_phone_headers, ['101A', '407-123-1234', 'NULL', '2019-09-06T13:00:00.000']),
+      CSV::Row.new(mental_phone_headers, ['202A', '321-123-1234', '0', '2019-09-06T13:00:00.000'])
+    ]
+
+    allow_any_instance_of(
+      Facilities::MentalHealthReloadJob
+    ).to receive(:fetch_mental_health_data).and_return(CSV::Table.new(mental_health_data))
+    Facilities::MentalHealthReloadJob.new.perform
+
+    expect(FacilityMentalHealth.find('101A')).to_not be_nil
+    expect(FacilityMentalHealth.find('202A')).to_not be_nil
+    expect(FacilityMentalHealth.find('101A').mh_ext).to eq(nil)
+    expect(FacilityMentalHealth.find('202A').mh_ext).to eq(nil)
+  end
+
+  context 'on error' do
```

Comment:
When describing a context, start its description with "when" or "with".

```suggestion
  context 'when encountering an error' do
```